src/Common/MemoryTracker.cpp

```diff
@@ -606,3 +614,20 @@ bool canEnqueueBackgroundTask()
     auto amount = background_memory_tracker.get();
     return limit == 0 || amount < limit;
 }
+
+void MemoryTracker::updateMemoryCredits()
+{
+    static thread_local Stopwatch stopwatch;
```

Comment:
If memory is allocated in one thread and released in another, it will be calculated incorrectly.

Reply:
Put Stopwatch inside MemoryTracker as a member. But you don't need "previous value" as a member, because every method like alloc/free already knows it.